src/main/java/org/grouplens/grapht/Instantiators.java

```diff
@@ -132,21 +144,25 @@ public Class getType() {
 
     private static class ProviderInstantiator implements Instantiator {
         private final Instantiator providerInstantiator;
-
         public ProviderInstantiator(Instantiator prov) {
             providerInstantiator = prov;
         }
 
         @Override
         public Object instantiate() throws ConstructionException {
+            Logger.trace(" provider invoked "+providerInstantiator);
```

Comment:
A few changes here, for style:

- let's say "invoking provider"
- no need for space at beginning
- let's use SLF4J interpolation instead of concatination

```
logger.trace("invoking provider {}", providerInstantiator);
```

Also, we should really log the provider, rather than its instantiator; so let's move this after the next line, and log `provider` instead of `providerInstantiator`.